### PR TITLE
Add spider for Rr Boa Vista

### PR DESCRIPTION
**Pull request by weibemoura**

### CODE REVIEW

File: processing/data_collection/gazette/spiders/rr_boa_vista.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        div_list = response.xpath('//*[@class="bldownload"]')
+        for div in div_list:
+            content = div.xpath("./div/text()").extract()
+            date = parse(content[1], languages=["pt"]).date()
+
+            url = div.xpath("./a/@href").extract_first()
+            url = response.urljoin(url)
+
+            power = "executive_legislature"
+            items.append(
```
Comment (rennerocha) [inline processing/data_collection/gazette/spiders/rr_boa_vista.py]: You don't need to include all the gazettes in a list before yielding it. Remove the `items` list and yield each item as they are scraped:
 ```
                yield Gazette(
                    date=date,
                    file_urls=[url],
                    is_extra_edition=False,
                    territory_id=self.TERRITORY_ID,
                    power=power,
                    scraped_at=dt.datetime.utcnow(),
                ))
```

File: processing/data_collection/gazette/spiders/rr_boa_vista.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+                Gazette(
+                    date=date,
+                    file_urls=[url],
+                    is_extra_edition=False,
+                    territory_id=self.TERRITORY_ID,
+                    power=power,
+                    scraped_at=dt.datetime.utcnow(),
+                ))
+
+        return items
```
Comment (rennerocha) [inline processing/data_collection/gazette/spiders/rr_boa_vista.py]: Remove this.

File: processing/data_collection/gazette/spiders/rr_boa_vista.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    def parse(self, response):
+        options = response.xpath('//*[@id="Periodo"]/optgroup/option/@value')
+        for option in options:
+            data = option.extract()
+
+            url = "".join([self.start_urls[0], "?Periodo=", data])
+            yield scrapy.Request(url, self.parse_period)
+
+    def parse_period(self, response):
+        items = []
```
Comment (rennerocha) [inline processing/data_collection/gazette/spiders/rr_boa_vista.py]: Remove this.

File: processing/data_collection/gazette/spiders/rr_boa_vista.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    name = "rr_boa_vista"
+    allowed_domains = ["boavista.rr.gov.br"]
+    start_urls = ["https://www.boavista.rr.gov.br/diario-oficial"]
+
+    def parse(self, response):
+        options = response.xpath('//*[@id="Periodo"]/optgroup/option/@value')
+        for option in options:
+            data = option.extract()
+
+            url = "".join([self.start_urls[0], "?Periodo=", data])
```
Comment (rennerocha) [inline processing/data_collection/gazette/spiders/rr_boa_vista.py]: Use `w3lib` to include this new parameter to the initial URL (http://w3lib.readthedocs.io/en/latest/w3lib.html#w3lib.url.add_or_replace_parameter)

```
import w3lib.url
url = w3lib.url.add_or_replace_parameter(response.url, 'Periodo', data)
```

**Comment by weibemoura** [comment]
Thanks @rennerocha I just updated the code

**Pull request merged**